src/tasks/add-solution-component/add-solution-component-v0/task.json

```diff
@@ -0,0 +1,95 @@
+{
+  "id": "783C008C-8AF1-4E47-B682-7A66FB64824C",
+  "name": "PowerPlatformAddSolutionComponent",
+  "friendlyName": "Power Platform Add Solution Component",
+  "description": "Adds a solution component to the target unmanaged solution in the connected Dataverse environment",
+  "author": "Microsoft",
+  "helpMarkDown": "[More Info](https://aka.ms/buildtoolsdoc)",
+  "category": "Deploy",
+  "visibility": [
+    "Build",
+    "Release"
+  ],
+  "demands": [
+    "PowerPlatformTool"
+  ],
+  "version": {
+    "Major": "0",
+    "Minor": "1",
+    "Patch": "0"
+  },
+  "preview": false,
+  "minimumAgentVersion": "1.95.0",
+  "instanceNameFormat": "Power Platform Add Solution Component $(message)",
+  "inputs": [
+    {
+      "name": "authenticationType",
+      "type": "radio",
+      "label": "Authentication type",
+      "defaultValue": "PowerPlatformEnvironment",
+      "required": true,
+      "helpMarkDown": "Authentication via username/password is simpler to set up but won't work if your AAD tenant requires MFA; use Service Principal Name (SPN) if MFA is required.",
+      "options": {
+        "PowerPlatformEnvironment": "Username/password (no MFA support)",
+        "PowerPlatformSPN": "Service Principal/client secret (supports MFA)"
```

Comment:
I know we've used similar verbiage elsewhere, but "supports MFA" isn't really accurate; the SPN auth goes a route that does not and cannot have MFA, so it permits orgs with MFA requirements to use the task by the fact that it is not using a standard user account.

I don't really have a suggestion for better terminology that communicates this without being overly verbose - "Service Principal/client secret (use this route if your organization requires MFA for standard user accounts as the Username/Password flow cannot be used)", but something we might want to but more thought into to clarify what the settings do in the future.

Reply:
That is true and could lead to confusion down the road.